tests/acceptance/10_files/01_create/cfengine_create_by_default.cf

```diff
@@ -0,0 +1,109 @@
+##############################################################################
+#
+# template_method cfengine should create files by default, unless
+# `create => "false"` is specified.
+#
+##############################################################################
+
+body common control
+{
+  inputs => { "../../default.cf.sub" };
+  bundlesequence  => { default("$(this.promise_filename)") };
+  version => "1.0";
+}
+
+##############################################################################
+
+bundle agent init
+{
+  vars:
+      "range"
+        slist => { expandrange("[1-4]", 1) };
+
+  files:
+      "$(G.testfile).test_$(range)"
+        delete => tidy;
+
+      "$(G.testfile).valid_template"
+        content => "[%CFEngine BEGIN %]
+I have $(const.dollar)(sys.cpus) CPU's!
+[%CFEngine END %]
+";
+
+      "$(G.testfile).invalid_template"
+        content => "[%CFEngine BEGIN %]
+I have $(const.dollar)(sys.cpus) CPU's!
+[%CFEngine END %)
+";
+}
+
+##############################################################################
+
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-3955" }
+        string => "template_method cfengine creates promiser by default";
+
+  files:
+    # File should be created by default
+      "$(G.testfile).test_1"
+        template_method => "cfengine",
+        edit_template => "$(G.testfile).valid_template",
+        if => fileexists("$(G.testfile).valid_template");
+
+    # File should not be created due to invalid template
+      "$(G.testfile).test_2"
+        template_method => "cfengine",
+        edit_template => "$(G.testfile).invalid_template",
+        if => fileexists("$(G.testfile).invalid_template");
+
+    # File should be created even though template is invalid
+      "$(G.testfile).test_3"
+        create => "true",
+        template_method => "cfengine",
+        edit_template => "$(G.testfile).invalid_template",
+        if => fileexists("$(G.testfile).invalid_template");
+
+    # File should not be created even though template is valid
+      "$(G.testfile).test_4"
+        create => "false",
+        template_method => "cfengine",
+        edit_template => "$(G.testfile).valid_template",
+        if => fileexists("$(G.testfile).valid_template");
```

Comment:
Shall we add a test with `create => "false"` and an existing file?

Reply:
Yes 😉